fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/ParameterVisitorBatchDAO.java

```diff
@@ -519,127 +472,29 @@ private void setLocationParms(PreparedStatement insert, int parameterNameId, dou
         insert.setDouble(2, lat);
         insert.setDouble(3, lng);
         insert.setLong(4, logicalResourceId);
+        setCompositeId(insert, 5);
+
     }
 
     @Override
     public void visit(CompositeParmVal compositeParameter) throws FHIRPersistenceException {
-        String parameterName = compositeParameter.getName();
+        if (this.currentCompositeId != null) {
+            // no soup for you
+            logger.warning("A compositeParameter '" + currentCompositeParameterName + "' cannot itself contain a composite '" + compositeParameter.getName());
+            throw new FHIRPersistenceException("composite parameters cannot themselves contain composites");
```

Comment:
Is this ever the case where composite contains a composite? I  don't think so, but verbailizing to confirm

Reply:
I mentioned in the call on Friday. I haven't found any, and didn't see any use-cases where this would be needed. The same logical behavior could be achieved with a separate composite parameter definition (after expanding the inner composite into its primitives) - and the good news is we are no longer limited to 3 parameters.